doc/semantics-of-the-publishing-api.md

```diff
@@ -0,0 +1,406 @@
+
+- *republish* - for when the content item needs to be re-sent to downstream
+systems
+
+- *links* - for when only the links of the content item have changed
+
+An example case for when a republish `update_type` should be used is when there
+had previously been a problem with a downstream system and that system did not
+act on publish events correctly. For example, there could have been a problem
+with sending email alerts and so one way that this could be rectified might be
+to republish the content items.
+
+The *links* `update_type` is set automatically when the /links endpoint is used.
+There is no need to set this manually. All of these `update_types` form part of
+the routing key when the content item document is placed on the message queue,
+together with the `format` of the content item (e.g. *policy.major*)
+
+---
+
+### **Optional Fields**
+
+---
+
+### description
+
+Example: *VAT rates for goods and services*
+
+Required: No
+
+The `description` captures a short explanation of the content item. It is sent
+downstream to the content store and may be displayed at the discretion of the
+front-end application, when the content item is rendered.
+
+---
+
+### need_ids
+
+Example: *["1234", "1235"]*
+
+Required: No
+
+The `needs_ids` are the identifiers of [user needs](https://www.gov.uk/design-principles)
+that are entered through the [Maslow Application](https://github.com/alphagov/maslow).
+They are passed through to the content store, untouched by the pipeline. The
+front-end applications can then use the `needs_ids` to present pages to users,
+such as [this one](https://www.gov.uk/info/overseas-passports).
+
+---
+
+### locale
+
+Example: *en*
+
+Required: No
+
+The `locale` is the language in which the content item is written. Front-end
+applications will optionally provide this string when querying the content store
+in order to retrieve content items in a given `locale`.
+
+A list of valid locales can be viewed in the Publishing API [here](https://github.com/alphagov/publishing-api/blob/master/config/application.rb#L32-L37).
+This field is not required and if a `locale` is not provided, it will be set to
+*en* automatically.
+
+---
+
+### phase
+
+Examples: *alpha, beta, live*
+
+Required: No
+
+The `phase` is an optional field that may be used to indicate the ‘Service
+Design Phase’ of the content item. The phase must be one of either *alpha*,
+*beta* or *live*. If the `phase` is not specified, the Publishing API will
+default it to *live*.
+
+Content items will be published to the content store regardless of their
+`phase`.  If a content item has a `phase` of either *alpha* or *beta*, a visual
+component will be added to the front-end to show this (assuming the front-end
+application for that format supports this).
+
+There is more information on what each of the phases mean [here](https://www.gov.uk/service-manual/phases).
+
+---
+
+### access_limited
+
+Example: *{ users: ["bf3e4b4f-f02d-4658-95a7-df7c74cd0f50"] }*
```

Comment:
I took this from the PACT: https://pact-broker.dev.publishing.service.gov.uk/pacts/provider/Content%20Store/consumer/Publishing%20API/latest